src/components/views/SupplierLookup.tsx

```diff
@@ -0,0 +1,184 @@
+import '../../styles/supplierLookup.css';
+import { Textfield } from '../base/Textfield';
+import '../../styles/globalbtn.css';
+import { useState } from 'react';
+import { Supplier } from '../data/data';
+
+import SupplierTable from '../inputs/SupplierTable';
+
+interface SupplierLookupProps {
+  onClose: () => void;
+  onSupplierSelect: (supplier: Supplier) => void;
+}
+
+const SupplierLookup: React.FC<SupplierLookupProps> = ({
+  onClose,
+  onSupplierSelect,
+}) => {
+  const [name, setName] = useState('');
+  const [supplierIndex, setSIndex] = useState<number | null>(null);
+  const [city, setCity] = useState('');
+  const [filteredSuppliers, setFilteredSuppliers] = useState<Supplier[]>([]);
+  const [selectedSupplierIndex, setSelectedSupplierIndex] = useState<
+    number | null
+  >(null);
+
+  const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setName(event.target.value);
+  };
+
+  const handleIndexChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSIndex(event.target.value ? parseInt(event.target.value) : null);
+  };
+
+  const handleCityChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setCity(event.target.value);
+  };
+
+  const handleSearch = (event: React.FormEvent) => {
+    event.preventDefault();
+    const hardcodedSuppliers: Supplier[] = [
```

Comment:
No right place to hardcode suppliers. 
1. any constant that do not depents on component props/states or any other fields should be out of the component
2. This should be "hardcoded" inserted in database, and lookup is doing search in database, not filtering hardcoded array